src/game/file_io.c

```diff
@@ -891,36 +870,31 @@ static int read_compressed_chunk(FILE *fp, void *buffer, int bytes_to_read, int
 
 static int zlib_compress(const void *input_buffer, int input_length, void *output_buffer, int *output_length)
 {
-    int ret, flush;
+    int ret;
     unsigned have;
     z_stream strm;
 
-    /* allocate deflate state */
     strm.zalloc = Z_NULL;
     strm.zfree = Z_NULL;
     strm.opaque = Z_NULL;
     ret = deflateInit(&strm, Z_BEST_SPEED);
-    if (ret != Z_OK)
-        return ret;
+    if (ret != Z_OK) {
+        return 0;
+    }
 
-    /* compress until end of file */
     strm.avail_in = input_length;
-    flush = Z_FINISH;
     strm.next_in = input_buffer;
     strm.avail_out = COMPRESS_BUFFER_SIZE;
     strm.next_out = output_buffer;
-    ret = deflate(&strm, flush);    /* no bad return value */
-    assert(ret != Z_STREAM_ERROR);  /* state not clobbered */
-    have = COMPRESS_BUFFER_SIZE - strm.avail_out;
-    assert(strm.avail_in == 0);     /* all input will be used */
-
-    assert(ret == Z_STREAM_END);        /* stream will be complete */
+    ret = deflate(&strm, Z_FINISH);
+    deflateEnd(&strm);
+    if (ret != Z_STREAM_END || strm.avail_in != 0) {
+        return 0;
+    }
 
+    have = COMPRESS_BUFFER_SIZE - strm.avail_out;
```

Comment:
You don't really need the `have`variable, you can use `*output_length = COMPRESS_BUFFER_SIZE - strm.avail_out;`